x-pack/plugins/observability_solution/synthetics/server/routes/default_alerts/default_alert_service.test.ts

```diff
@@ -0,0 +1,466 @@
+      return {
+        ...omit(mockAlert, 'systemActions'),
+        actions: [...mockAlert.actions, ...mockAlert.systemActions],
+      };
+    }
+
+    afterEach(() => jest.resetAllMocks());
+
+    describe('getExistingAlert', () => {
+      it('returns alert if exists', async () => {
+        const { getRulesClient, mockAlert } = setUpExistingAlerts();
+        const service = new DefaultAlertService(
+          { alerting: { getRulesClient } } as any,
+          {} as any,
+          {} as any
+        );
+        const result = await service.getExistingAlert('xpack.synthetics.alerts.monitorStatus');
+        expect(result).toEqual(formatMockAlertResult(mockAlert));
+      });
+
+      it('returns undefined if alert does not exist', async () => {
+        const find = jest.fn().mockResolvedValue({ data: [] });
+        const getRulesClient = jest.fn();
+        getRulesClient.mockReturnValue({ find });
+        const service = new DefaultAlertService(
+          { alerting: { getRulesClient } } as any,
+          {} as any,
+          {} as any
+        );
+        const result = await service.getExistingAlert('xpack.synthetics.alerts.monitorStatus');
+        expect(result).toBeUndefined();
+      });
+    });
+    describe('createDefaultAlertIfNotExist', () => {
+      it('returns alert if exists', async () => {
+        const { getRulesClient, mockAlert } = setUpExistingAlerts();
+        const service = new DefaultAlertService(
+          { alerting: { getRulesClient } } as any,
+          {} as any,
+          {} as any
+        );
+        const alert = await service.createDefaultAlertIfNotExist(
+          'xpack.synthetics.alerts.monitorStatus',
+          'name',
+          '1m'
+        );
+        expect(alert).toEqual(formatMockAlertResult(mockAlert));
+        expect(getRulesClient).toHaveBeenCalled();
+      });
+
+      it('creates alert if does not exist', async () => {
+        const sampleAction = { alertsFilter: { query: { kql: 'some kql', filters: [] } } };
+        const find = jest.fn().mockResolvedValue({ data: [] });
+        const create = jest.fn().mockResolvedValue({
+          actions: [sampleAction],
+          systemActions: [],
+          id: '123',
+          alertTypeId: 'testalertid',
+        });
+        const getActionsClient = jest.fn();
+        getActionsClient.mockReturnValue({
+          getAll: jest
+            .fn()
+            .mockResolvedValue([{ id: 'id', actionTypeId: 'actionTypeId', name: 'action name' }]),
+        });
+        const getRulesClient = jest.fn();
+        getRulesClient.mockReturnValue({ find, create });
+        const service = new DefaultAlertService(
+          { actions: { getActionsClient }, alerting: { getRulesClient } } as any,
+          {} as any,
+          {} as any
+        );
+        service.settings = { defaultConnectors: ['slack', 'email'] } as any;
+        const result = await service.createDefaultAlertIfNotExist(
```

Comment:
Can we rename this to `createDefaultRuleIfDoesNotExist` and update the test to use the word rule instead of alert as well.

Reply:
++ agree, this is a case of perpetuating a bad habit

Reply:
see https://github.com/elastic/kibana/pull/193235/commits/420a7b51a74689b726a3824a881e83b71257a436